### PR TITLE
Adding support for idleTimeout in Service Router spec

### PR DESCRIPTION
**Pull request by kphatak**
Changes proposed in this PR:
- Adding support for providing idleTimeout in the Service Router config.
- Addressing https://github.com/hashicorp/consul-k8s/issues/2153

Contributing for the first time on consul-k8s. Followed [guide](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#making-changes-to-consul-k8s)

How I've tested this PR:
Not sure what I need to run locally to validate template changes.

How I expect reviewers to test this PR:

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: control-plane/api/v1alpha1/servicerouter_types.go [hunk 1]
```diff
@@ -142,6 +142,9 @@ type ServiceRouteDestination struct {
 	PrefixRewrite string `json:"prefixRewrite,omitempty"`
 	// RequestTimeout is the total amount of time permitted for the entire
 	// downstream request (and retries) to be processed.
+	IdleTimeout metav1.Duration `json:"idleTimeout,omitempty"`
```
[the remark this answers is not in the document]
Comment (kphatak) [reply control-plane/api/v1alpha1/servicerouter_types.go:145]: I think that's what I was looking for. Thank you. Let me make the changes.

[the remark this answers is not in the document]
Comment (kphatak) [reply control-plane/api/v1alpha1/servicerouter_types.go:145]: @thisisnotashwin When I run `make ctrl-generate ctrl-manifests`, it swaps the descriptions for IdleTimeout and RequestTimeout. I am not sure why that is happening 🤔

[the remark this answers is not in the document]
Comment (kphatak) [reply control-plane/api/v1alpha1/servicerouter_types.go:145]: It is also reverting header/copyright changes made by this commit
https://github.com/hashicorp/consul-k8s/commit/85c28bd4d76c6f8539060fdd8e1d8a90716becb4

[the remark this answers is not in the document]
Comment (thisisnotashwin) [reply control-plane/api/v1alpha1/servicerouter_types.go:145]: > it swaps the descriptions for IdleTimeout and RequestTimeout
you will want to swap the contents of L143-144 with L146-147. The description of the field should be above the field.

> reverting header/copyright changes made by this commit
We havent fixed the templating to that yet, unfortunately. Can you ignore the header changes that get made from the commit?

**Review by thisisnotashwin — APPROVED** [review]
Thanks so much for the contribution!! this looks really good!

**Comment by kphatak** [comment]
Thank you. How do I backport it to 1.14.x as the feature was added in 1.14?

**Review by curtbushko — APPROVED** [review]
LGTM

**Comment by david-yu** [comment]
Roughly speaking it's around the beginning of July, we have 1.16 GA coming this month so we need to get to follow up patch releases afterwards.

**Pull request merged**